backend/src/users/user.service.ts

```diff
@@ -7,17 +7,35 @@ import * as bcrypt from 'bcrypt';
 
 @Injectable()
 export class UsersService {
-  constructor(
-    @InjectModel(User.name) private readonly userModel: Model<UserDocument>,
-  ) {}
+    constructor(
+        @InjectModel(User.name) private userModel: Model<UserDocument>,
+    ) {}
 
-  async create(createUserDto: CreateUserDto): Promise<User> {
-    createUserDto.password = await bcrypt.hash(createUserDto.password, 8)
-    const createdUser = new this.userModel(createUserDto);
-    return createdUser.save();
-  }
+    async create(createUserDto: CreateUserDto): Promise<User> {
+        createUserDto.password = await bcrypt.hash(createUserDto.password, 8)
+        createUserDto.token = await bcrypt.hash(createUserDto.password, 7)
+        const createdUser = new this.userModel(createUserDto);
+        return createdUser.save();
+    }
 
-  async findAll(): Promise<User[]> {
-    return this.userModel.find().exec();
-  }
+    async findAll(): Promise<User[]> {
+        return this.userModel.find().exec();
+    }
+
+    async findOne(@Param(':id') id) {
+        const users = await this.userModel.find().exec()
+        const user = users.filter(user => user._id == id)
```

Comment:
What is the reason for using double equals here instead of triple equals?

Reply:
Actually, it doesn't work with triple equals.

Reply:
MongoDB `_id` is of type ObjectId:
https://docs.mongodb.com/manual/reference/method/ObjectId/

You can call `.toString()` method on it, then compare it to the `id` parameter using triple equals.

Reply:
Okay, it works now.